sql: prevent REGIONAL BY ROW and ADD/DROP REGION concurrent transitions

otan opened the pull request:
See individual commits for details.

Refs: #63368

otan committed:
sql: prevent region configuration during REGIONAL BY ROW transitions

Release note (sql change): Disallow ALTER DATABASE ... ADD|DROP|SET
PRIMARY REGION during REGIONAL BY ROW table transitions.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/63460)
<!-- Reviewable:end -->

otan committed:
sql: prevent REGIONAL BY ROW transition during ADD/DROP region

Release note (sql change): Prevent REGIONAL BY ROW transitions when an
ADD/DROP REGION is also bieng performed at the same time.
